Prune metrics: others 11/DoNe

Borda opened the pull request:
## What does this PR do?

Prune remaining metrics
follow-up #6505

## Before submitting
- [ ] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

<!-- For CHANGELOG separate each item in the unreleased section by a blank line to reduce collisions -->

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
Make sure you had fun coding 🙃

carmocca commented on pytorch_lightning/metrics/functional/classification.py:
```diff
@@ -223,32 +169,8 @@ def precision(
-        tensor(0.7500)
-
+    .. deprecated::
+        Use :func:`torchmetrics.functional.precision_recall`. Will be removed in v1.4.0.
```
```suggestion
        Use :func:`torchmetrics.functional.precision`. Will be removed in v1.4.0.
```

carmocca commented on pytorch_lightning/metrics/functional/classification.py:
```diff
@@ -267,31 +189,8 @@ def recall(
-        >>> recall(x, y)
-        tensor(0.7500)
+    .. deprecated::
+        Use :func:`torchmetrics.functional.precision_recall`. Will be removed in v1.4.0.
```
```suggestion
        Use :func:`torchmetrics.functional.recall`. Will be removed in v1.4.0.
```

carmocca commented on pytorch_lightning/metrics/functional/classification.py:
```diff
@@ -427,68 +291,15 @@ def multiclass_auroc(
-        >>> multiclass_auroc(pred, target, num_classes=4)
-        tensor(0.6667)
+    .. deprecated::
+        Use :func:`torchmetrics.functional.auroc`. Will be removed in v1.4.0.
```
```suggestion
        Use :func:`torchmetrics.functional.multiclass_auroc`. Will be removed in v1.4.0.
```

Borda replied:
TM does not have this one...

rohitgr7 commented on pytorch_lightning/metrics/functional/classification.py:
```diff
@@ -73,16 +55,9 @@ def stat_scores_multiple_classes(
-        ValueError:
-            If ``reduction`` is not one of ``"none"``, ``"sum"`` or ``"elementwise_mean"``.
+    .. deprecated::
+        Use :func:`torchmetrics.functional.stat_scores`. Will be removed in v1.4.0.
```
```suggestion
        Use :func:`torchmetrics.functional.stat_scores_multiple_classes`. Will be removed in v1.4.0.
```

Borda replied:
this is not a valid func in `torchmetrics`

rohitgr7 reviewed (approved):
quick question: why some of them will be removed in v1.4 and others in v1.5?
here it's mentioned v1.5: https://pytorch-lightning.readthedocs.io/en/latest/extensions/metrics.html

Borda commented:
> quick question: why some of them will be removed in v1.4 and others in v1.5?

as you can see this marked for v1.4 removal was moved internally already for v1.2
so there are kind of two waves of refactoring>>deprecation
1. we moved some functioning from general classification.py to their one modules
2. we moved all to `torchmetrics` package
